OrchardCore.Commerce/Views/ShoppingCart/Index.cshtml

```diff
@@ -21,30 +20,36 @@
             <div class="col-xs-12 col-sm-1">@T["Price"]</div>
             <div class="col-xs-12 col-sm-1">@T["Action"]</div>
         </div>
-        @{
-            int line = 0;
-        }
-        @foreach (ShoppingCartLineViewModel item in Model.Lines)
+        @for (var line = 0; line < Model.Lines.Count; line++)
         {
+            var item = Model.Lines[line];
+            var name = $"cart.lines[{line}]";
+            var attributes = item
+                .Attributes
+                .Select((attribute, index) => (attribute.Value, Type: attribute.GetType().Name, Index: index))
+                .Where(pair => pair.Value.UntypedValue != null)
```

Comment:
Are three still a pair? 😁 Use `tuple` instead.

Reply:
ok, now I want to rename it to `crowd` 😁